lib/rbi/rewriters/replace_attributes_with_methods.rb

```diff
@@ -0,0 +1,146 @@
+# typed: strict
+# frozen_string_literal: true
+
+module RBI
+  module TempHelpers
+    extend T::Sig
+  end
+
+  module Rewriters
+    class ReplaceAttributesWithMethods < Visitor
+      extend T::Sig
+      include TempHelpers
+
+      sig { override.params(node: T.nilable(Node)).void }
+      def visit(node)
+        return unless node
+
+        case node
+        when Tree
+          node.nodes.dup.each do |child|
+            visit(child)
+            next unless (attr = child).is_a?(Attr)
+
+            new_methods = convert_to_methods(attr)
+
+            child.replace_with_multiple(new_methods)
+          end
+        end
+      end
+
+      private
+
+      sig { params(attr: Attr).returns(T::Array[Method]) }
+      def convert_to_methods(attr)
+        sig, attribute_type = parse_sig_of(attr)
+
+        case attr
+        when AttrReader then convert_attr_reader_to_methods(attr, sig, attribute_type)
+        when AttrWriter then convert_attr_writer_to_methods(attr, sig, attribute_type)
+        when AttrAccessor then convert_attr_accessor_to_methods(attr, sig, attribute_type)
+        else raise NotImplementedError, "Unknown attribute type: #{attr.class}"
+        end
+      end
+
+      sig { params(attr: AttrReader, sig: T.nilable(Sig), attribute_type: T.nilable(String)).returns(T::Array[Method]) }
+      def convert_attr_reader_to_methods(attr, sig, attribute_type)
+        attr.names.map do |name|
+          create_getter_method(name.to_s, sig, attr.visibility, attr.loc, attr.comments)
+        end
+      end
+
+      sig { params(attr: AttrWriter, sig: T.nilable(Sig), attribute_type: T.nilable(String)).returns(T::Array[Method]) }
+      def convert_attr_writer_to_methods(attr, sig, attribute_type)
+        attr.names.map do |name|
+          create_setter_method(name.to_s, sig, attribute_type, attr.visibility, attr.loc, attr.comments)
+        end
+      end
+
+      sig do
+        params(attr: AttrAccessor, sig: T.nilable(Sig), attribute_type: T.nilable(String)).returns(T::Array[Method])
+      end
+      def convert_attr_accessor_to_methods(attr, sig, attribute_type)
+        readers = attr.names.flat_map do |name|
+          create_getter_method(name.to_s, sig, attr.visibility, attr.loc, attr.comments)
+        end
+
+        writers = attr.names.map do |name|
+          create_setter_method(name.to_s, sig, attribute_type, attr.visibility, attr.loc, attr.comments)
+        end
+
+        readers + writers
+      end
+
+      sig { params(attr: Attr).returns([T.nilable(Sig), T.nilable(String)]) }
+      def parse_sig_of(attr)
+        raise "Attributes cannot have more than 1 sig" if 1 < attr.sigs.count
```

Comment:
I would rather not have RBI do this check, as far as we know this is perfectly valid RBI syntax, the semantics of it should be left to Tapioca/Sorbet